solutions/fscloud/main.tf

```diff
@@ -0,0 +1,48 @@
+########################################################################################################################
+# Resource Group & Service ID
+########################################################################################################################
+
+module "resource_group" {
+  source  = "terraform-ibm-modules/resource-group/ibm"
+  version = "1.1.4"
+  # if an existing resource group is not set (null) create a new one using prefix
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+
+resource "ibm_iam_service_id" "resource_keys_existing_serviceids" {
+  count       = 3
+  name        = "${var.prefix}-serviceid-${count.index}"
+  description = "ServiceID for ${var.prefix} env to use for resource key credentials"
+}
+
+########################################################################################################################
+# AppID instance
+########################################################################################################################
+
+module "appid" {
+  source                     = "../../modules/fscloud"
+  appid_name                 = "${var.prefix}-appid"
+  resource_group_id          = module.resource_group.resource_group_id
+  region                     = var.region
+  resource_tags              = var.resource_tags
+  kms_key_crn                = var.kms_key_crn
+  existing_kms_instance_guid = var.existing_kms_instance_guid
+  resource_keys = [{
```

Comment:
The DA should not be creating all these resource keys by default. Its up to the consumer if they want to create resource keys or not. Whatever approach we go with here, we should be consistent with other DAs. One option could be to expose 3 boolean variables like `create_writer_resource_key`, `create_manager_resource_key` and `create_writer_resource_key`? The other option is to expose the `resource_keys` variable, but we kind of want to avoid exposing object type vars in DAs as they are hard for consumers to pass values to in Projects / Schematics.

Reply:
Using approach #2